File: src/test/java/org/interview/jobtask/shortener/URLStorageImplTest.java
package org.interview.jobtask.shortener;

import org.interview.jobtask.shortener.storage.KeywordCollisionException;
import org.interview.jobtask.shortener.storage.URLStorageImpl;
import org.junit.Test;

public class URLStorageImplTest {

    private URLStorageImpl storage = URLStorageImpl.instance();

    @Test(expected = KeywordCollisionException.class)
    public void store() throws KeywordCollisionException {
        storage.store("https://blog.mysite.com/cool-article", "BEST-ARTICLE");
        storage.store("https://blog.mysite.com/any", "BEST-ARTICLE");
    }

}

File: src/test/kotlin/org/interview/jobtask/shortener/storage/kt/URLStorageImplTest.kt
package org.interview.jobtask.shortener.storage.kt

import org.interview.jobtask.shortener.org.interview.jobtask.shortener.storage.kt.URLStorageImpl
import org.interview.jobtask.shortener.storage.KeywordCollisionException
import org.junit.Before
import org.junit.Test
import kotlin.test.assertEquals

class URLStorageImplTest {

    private val storage =  URLStorageImpl.instance()

    @Before
    fun before() {
        storage["BEST-ARTICLE"] = "https://blog.mysite.com/cool-article"
    }

    @Test(expected = KeywordCollisionException::class)
    fun store() {
        storage["BEST-ARTICLE"] = "https://blog.mysite.com/any"
    }

    @Test
    fun retrieve() {
        assertEquals("https://blog.mysite.com/cool-article", storage["BEST-ARTICLE"])
    }
}